osbs/api.py

```diff
@@ -395,18 +400,26 @@ def _create_build_config_and_build(self, build_request):
             logger.debug('Changed parent ImageStreamTag? %s', changed_ist)
 
         if triggers:
-            existing_bc['spec']['triggers'] = triggers
-            self.os.update_build_config(build_config_name, json.dumps(existing_bc))
+            patch_triggers = {'spec': {'triggers': None}}
+            self.os.patch_build_config(build_config_name, json.dumps(patch_triggers))
+            patch_triggers = {'spec': {'triggers': triggers}}
+            self.os.patch_build_config(build_config_name, json.dumps(patch_triggers))
 
         if image_stream and triggers:
-            prev_version = existing_bc['status']['lastVersion']
             build_id = self.os.wait_for_new_build_config_instance(
-                build_config_name, prev_version)
+                build_config_name, existing_bc['status']['lastVersion'])
             build = BuildResponse(self.os.get_build(build_id).json())
         else:
             response = self.os.start_build(build_config_name)
             build = BuildResponse(response.json())
 
+        if orchestrator_build:
```

Comment:
Is it because the orchestrator is the only one that will have triggers? Would a better test be whether the BuildConfig has triggers?